JS/Lesson18/goksel/script.js

```diff
@@ -27,17 +36,27 @@ console.log('----------------------------');
 // 3. Write a function that takes an array of strings and concatenates them together adding spaces.
 // Input: ["Hello", "World", "!"]
 // Output: "Hello World !"
-let stringEx3 = ["Hello", "World", "!"];
+let stringEx3 = ["Hello", "World", "!", "I am", "web developer", "?"];
 
 function concatenateStrings(array) {
   let sentence = '';
   // TODO: if the next character is one of ?, '.', ',', etc. do not add a space before that character
   for (let i = 0; i < array.length; i++) {
-    sentence = sentence + ' ' + array[i];
+    const specialCharacter = ["?",".",",","!",";",":","'","-","_","="];
+    const nextArray = array[i+1];
```

Comment:
```suggestion
    const nextArrayCharacter = array[i+1];
```

Using the name "nextArray" makes the person reading the code assume that the value is an array which is not true.

Reply:
Also, the other issue here is that you're checking the word that comes after the current one however you need to check the current word instead. If you check the current word, then the logic will work as you think it should:
```
let stringEx3 = ["Hello", "World", "!", "I am", "web developer", "?"];

function concatenateStrings(array) {
  let sentence = '';
  // TODO: if the next character is one of ?, '.', ',', etc. do not add a space before that character
  for (let i = 0; i < array.length; i++) {
    const specialCharacter = ["?",".",",","!",";",":","'","-","_","="];
    const nextArray = array[i];
    if(!specialCharacter.includes(nextArray)){
      sentence = sentence + " " + array[i];
    }else {
      sentence += array[i];
    }
  }
  return sentence; 
};
```

Reply:
function concatenateStrings(array) {
let sentence = "" ;
for (let i = 0; i < array.length; i++){
const specialCharacter = ["?",".",",","!",";",":","'","-","_","="];
const newtArrayCharacter = array[i];
if(!specialCharacter.includes(newtArrayCharacter)){
sentence = sentence + " " + array[i];
}else {
sentence = sentence + array[i];
}
}
return sentence;
}